overlord/snapstate/target.go

```diff
@@ -796,6 +796,62 @@ func (p *updatePlan) targetInfos() []*snap.Info {
 	return infos
 }
 
+// updates returns the updates that should be applied to the system's state for
+// this plan.
+func (p *updatePlan) updates(st *state.State, opts Options) ([]update, error) {
+	updates := make([]update, 0, len(p.targets))
+	for _, t := range p.targets {
+		opts.PrereqTracker.Add(t.info)
+
+		snapsup, compsups, err := t.setups(st, opts)
+		if err != nil {
+			if !p.refreshAll() {
+				return nil, err
+			}
+
+			logger.Noticef("cannot refresh snap %q: %v", t.info.InstanceName(), err)
+			continue
+		}
+
+		updates = append(updates, update{
+			Setup:      snapsup,
+			SnapState:  t.snapst,
+			Components: compsups,
+		})
+	}
+	return updates, nil
+}
+
+// revisionChanges returns the snaps that will have their revisions changed by
+// the updates in this plan.
+func (p *updatePlan) revisionChanges(st *state.State, opts Options) ([]*snap.Info, error) {
```

Comment:
nitpick is there code that this and doUpdate could share?

Reply:
`doUpdate` unfortunately doesn't operate on an `updatePlan`, since it holds onto some `snap.Info` structs, which we don't have during a resuming an inhibited refresh.